primitives/runtime/Cargo.toml

```diff
@@ -15,7 +15,7 @@ targets = ["x86_64-unknown-linux-gnu"]
 
 
 [dependencies]
-serde = { version = "1.0.101", optional = true, features = ["derive"] }
+serde = { version = "1.0.101", default-features = false, features = ["derive"] }
```

Comment:
The main reason of not implementing JSON-RPC requests in the runtime was to avoid bringing in `serde` and `serde_json` due to potential runtime size increase.

I remember @Xanewok was doing some tests on this, and afair surprisingly `serde_json` was not really increasing the runtime size that much compared to other json libs (like `simplejson`)?

@Xanewok can you confirm?

Reply:
Oh! I wasn't aware of runtime size issues. One of the reasons why I thought `serde_json` was a good choice is because it is used in one of the Substrate Cookbook recipes:

https://github.com/substrate-developer-hub/recipes/blob/master/pallets/ocw-demo/src/lib.rs

I see that alternatively, `lite-json` has been used as well:

https://github.com/paritytech/substrate/blob/master/frame/example-offchain-worker/src/lib.rs#L614

Reply:
Yeah, I think it should be good, but would prefer to have some confirmations. Ideally few measurements of how different libs affect the runtime size.

Reply:
> The main reason of not implementing JSON-RPC requests in the runtime was to avoid bringing in `serde` and `serde_json` due to potential runtime size increase.
> 
> I remember @Xanewok was doing some tests on this, and afair surprisingly `serde_json` was not really increasing the runtime size that much compared to other json libs (like `simplejson`)?
> 
> @Xanewok can you confirm?

Here's the data I pulled in April last year - then, the addition of introducing `serde_json` and serializing to an untyped `serde_json::Value` was almost none (when comparing some local code that did the manual string slicing to parse some simple JSOn value) but if we introduced `derive`s for `Serialize` and/or `Deserialize` on some struct definition, the runtime increased by quite a bit:
```
# $ cargo build --manifest-path=bin/node-template/runtime/Cargo.toml --release --target=wasm32-unknown-unknown --no-default-features
# $ ls -la target/wasm32-unknown-unknown/release/wbuild/node-template-runtime/node_template_runtime.compact.wasm | cut -d' ' -f5

# Baseline BTC pricefetch example (string slicing)
# https://github.com/xanewok/substrate/commit/2da23d5dd908a9e83b829e623cb032dfeca213ba
295174
# lite-json (to JSON value)
# https://github.com/xanewok/substrate/commit/1f03c5ac5a5f4649ac6f9cec8cf5fc5894772520
309536
# serde_json (to JSON value, with https://github.com/serde-rs/json/pull/588 applied, pulls alt_serde due to https://github.com/rust-lang/cargo/issues/4866)
# https://github.com/xanewok/substrate/commit/724ad7afb3616c4fb5f338ad91883c99becda5a7
291467
# serde_json (as above; deserializes to struct with alt_serde_derive)
# https://github.com/xanewok/substrate/commit/47b010aad101695d01bced51d1b705f77f35b1a1
316238
# as above, deserialize a dummy structure
# https://github.com/xanewok/substrate/commit/f574672c1178bba385b1e5d0e572b86d2fbb4045
326096
# impl serialize on both structs
# https://github.com/xanewok/substrate/commit/2e0cb27ad64aee5e2066aa319ce466cbea702e90
345010
```

It seems that there is no silver bullet, but using plain `serde_json` looks like a best solution (at least it did a year ago!).

Reply:
@Xanewok thanks!

@NachoPal so sorry for going back and worth, but can we avoid using `derive` and only use `serde_json` to serialize/deserialize the structs (I guess it means using `json!` macro and maybe deseiralizing to `Value` and then extracting the fields).

Reply:
Done